src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -179,53 +169,35 @@ def enrich_span_with_llm_if_needed(span, original_span, inputs, generator_output
             token_collector.collect_openai_tokens_for_streaming(span, inputs, generator_output, parser.is_chat)
 
 
-def traced_generator(original_span: ReadableSpan, inputs, generator):
-    context = original_span.get_span_context()
-    link = Link(context)
-    # If start_trace is not called, the name of the original_span will be empty.
-    # need to get everytime to ensure tracer is latest
-    otel_tracer = otel_trace.get_tracer("promptflow")
-    with otel_tracer.start_as_current_span(
-        f"Iterated({original_span.name})",
-        links=[link],
-    ) as span, _record_cancellation_exceptions_to_span(span):
-        enrich_span_with_original_attributes(span, original_span.attributes)
-        # Enrich the new span with input before generator iteration to prevent loss of input information.
-        # The input is as an event within this span.
-        enrich_span_with_input(span, inputs)
-        generator_proxy = GeneratorProxy(generator)
-        yield from generator_proxy
-        generator_output = generator_proxy.items
-        enrich_span_with_llm_if_needed(span, original_span, inputs, generator_output)
-        enrich_span_with_openai_tokens(span, TraceType(original_span.attributes["span_type"]))
-        enrich_span_with_output(span, serialize_attribute(generator_output))
-        span.set_status(StatusCode.OK)
+def traced_generator(span, inputs, generator, trace_type):
+    generator_proxy = GeneratorProxy(generator)
+    yield from generator_proxy
```

Comment:
catch exceptions to make sure the span could be ended even there are any errors?

Reply:
Sure.